### PR TITLE
Reconciler periodic upgrade test job

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Added job: `periodic-main-kyma-incubator-reconciler-kyma1-kyma2-upgrade`

The job steps are:
1.  Install Kyma latest release (e.g. 1.24.6) using cli.
2. Run fast-integration tests.
3. Reconicle kyma to main, using reconciler.
4. Run fast-integration tests again.

[Example Job Run](https://status.build.kyma-project.io/view/gs/kyma-prow-logs/logs/mfaizanse_test_of_prowjob_periodic-main-kyma-incubator-reconcil/1438052736212930560)

Prow: [Job History](https://status.build.kyma-project.io/job-history/gs/kyma-prow-logs/logs/mfaizanse_test_of_prowjob_periodic-main-kyma-incubator-reconcil)

**Related issue(s)**

https://github.com/kyma-incubator/reconciler/issues/75

### CODE REVIEW

File: prow/scripts/cluster-integration/reconciler-e2e-upgrade-gardener.sh [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    # shellcheck source=prow/scripts/lib/gardener/azure.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/azure.sh"
+elif [[ $GARDENER_PROVIDER == "aws" ]]; then
+    # shellcheck source=prow/scripts/lib/gardener/aws.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/aws.sh"
+elif [[ $GARDENER_PROVIDER == "gcp" ]]; then
+    # shellcheck source=prow/scripts/lib/gardener/gcp.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/gcp.sh"
+else
+    ## TODO what should I put here? Is this a backend?
```
Comment (sayanh) [inline prow/scripts/cluster-integration/reconciler-e2e-upgrade-gardener.sh:72]: is this TODO still valid?

File: prow/scripts/cluster-integration/reconciler-e2e-upgrade-gardener.sh [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+log::banner "Installing Kyma $KYMA_SOURCE"
+gardener::install_kyma
+
+# generate pod-security-policy list in json
+utils::save_psp_list "${ARTIFACTS}/kyma-psp.json"
+
+# run the fast integration test before reconciliation
+log::banner "Executing pre-upgrade test - before reconciliation"
+reconciler::pre_upgrade_test_fast_integration_kyma_1_24
+#gardener::test_fast_integration_kyma
```
Comment (sayanh) [inline prow/scripts/cluster-integration/reconciler-e2e-upgrade-gardener.sh:132]: unused command

File: prow/scripts/cluster-integration/reconciler-e2e-upgrade-gardener.sh [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+# Set up test pod environment
+reconciler::initialize_test_pod
+
+# Run a test pod from where the reconciliation will be triggered
+reconciler::reconcile_kyma
+
+# run the fast integration test after reconciliation
+log::banner "Executing post-upgrade test - after reconciliation"
+gardener::post_upgrade_test_fast_integration_kyma
+#gardener::test_fast_integration_kyma
```
Comment (sayanh) [inline prow/scripts/cluster-integration/reconciler-e2e-upgrade-gardener.sh:149]: lets enable and make it red

File: prow/scripts/cluster-integration/reconciler-e2e-upgrade-gardener.sh [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+# Wait until reconciler is ready
+reconciler::wait_until_is_ready
+
+# Install Kyma using cli with version previously set in KYMA_SOURCE
+log::banner "Installing Kyma $KYMA_SOURCE"
+gardener::install_kyma
+
+# generate pod-security-policy list in json
+utils::save_psp_list "${ARTIFACTS}/kyma-psp.json"
```
Comment (sayanh) [inline prow/scripts/cluster-integration/reconciler-e2e-upgrade-gardener.sh:127]: Why do we need this?

File: templates/data/incubator-buildpack-data.yaml [hunk 5]
```diff
@@ -230,6 +237,26 @@ templates:
...
+          reconciler_gardener_e2e_upgrade:
+            labels:
+              preset-gardener-gcp-kyma-integration: "true"
+              preset-kyma-cli-stable: "true"
+              preset-cluster-version: "true"
+            env:
+              KYMA_PROJECT_DIR: "/home/prow/go/src/github.com/kyma-project"
+              GARDENER_REGION: "europe-west4"
+              GARDENER_ZONES: "europe-west4-b"
+            annotations:
```
Comment (sayanh) [inline templates/data/incubator-buildpack-data.yaml:255]: lets use [gardener_gcp_job](https://github.com/kyma-project/test-infra/blob/main/templates/data/kyma-integration-gardener-data.yaml#L67) and avoid hard coding here

File: templates/data/incubator-buildpack-data.yaml [hunk 6]
```diff
@@ -230,6 +237,26 @@ templates:
...
+              - "/home/prow/go/src/github.com/kyma-incubator/reconciler"
+            branches:
+              - ^main$
+          reconciler_gardener_e2e_upgrade:
+            labels:
+              preset-gardener-gcp-kyma-integration: "true"
+              preset-kyma-cli-stable: "true"
+              preset-cluster-version: "true"
+            env:
+              KYMA_PROJECT_DIR: "/home/prow/go/src/github.com/kyma-project"
```
Comment (sayanh) [inline templates/data/incubator-buildpack-data.yaml:252]: lets use [gardener_gcp_job](https://github.com/kyma-project/test-infra/blob/main/templates/data/kyma-integration-gardener-data.yaml#L67) and avoid hard coding here

File: prow/scripts/cluster-integration/reconciler-e2e-upgrade-gardener.sh (hunk 1 above)
Comment (mfaizanse) [reply prow/scripts/cluster-integration/reconciler-e2e-upgrade-gardener.sh]: updated

File: prow/scripts/cluster-integration/reconciler-e2e-upgrade-gardener.sh (hunk 2 above)
Comment (mfaizanse) [reply prow/scripts/cluster-integration/reconciler-e2e-upgrade-gardener.sh]: removed

File: prow/scripts/cluster-integration/reconciler-e2e-upgrade-gardener.sh (hunk 3 above)
Comment (mfaizanse) [reply prow/scripts/cluster-integration/reconciler-e2e-upgrade-gardener.sh]: updated

File: prow/scripts/cluster-integration/reconciler-e2e-upgrade-gardener.sh (hunk 4 above)
Comment (mfaizanse) [reply prow/scripts/cluster-integration/reconciler-e2e-upgrade-gardener.sh]: removed

File: templates/data/incubator-buildpack-data.yaml (hunk 6 above)
Comment (mfaizanse) [reply templates/data/incubator-buildpack-data.yaml]: done

**Review by suleymanakbas91 — APPROVED** [review]

**Review by Ressetkk — APPROVED** [review]

**Pull request merged**